### PR TITLE
Add ProjectCentralUtils class

### LINKED ISSUE #31200: Provide a util API to pull a package from central
**Description:**
Prerequisite for #30757

### PR DESCRIPTION
**Pull request by Ibaqu**
## Purpose
Add class and resolve dependencies for pulling packages from central

Fixes #31200

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by pramodya1994** [comment]
@Ibaqu Shall we resolve the conflicts?

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+
+/**
+ * Utilities related to Central functionality.
+ *
+ * @since 2.0.0
+ */
+public class ProjectCentralUtils {
+
+    private static final String SUPPORTED_PLATFORM = "java11";
```
Comment (pramodya1994) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java:27]: Try to use `JvmTarget.JAVA_11.code()`

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                    .from(String.valueOf(settingsFilePath.getFileName()), Files.readString(settingsFilePath));
+            SettingsBuilder settingsBuilder = SettingsBuilder.from(settingsTomlDocument);
+            return settingsBuilder.settings();
+        } catch (IOException e) {
+            // If Settings.toml not exists return empty Settings object
+            return Settings.from();
+        }
+    }
+
+}
```
Comment (pramodya1994) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java:77]: We have same method in `CentralUtils`, Can't we use that?

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * Utilities related to Central functionality.
+ *
+ * @since 2.0.0
+ */
+public class ProjectCentralUtils {
+
+    private static final String SUPPORTED_PLATFORM = "java11";
+
+    public static boolean pullPackage(String orgName, String packageName, String version) throws ProjectException {
```
Comment (pramodya1994) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java:29]: Is there a possibility of moving this method to `CentralUtils`?

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+    }
+
+    /**
+     * Read Settings.toml to populate the configurations.
+     *
+     * @return {@link Settings} settings object
+     */
+    private static Settings readSettings() {
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java:64]: Let's use the one in CentralClient

Comment (Ibaqu) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java:64]: This causes a cyclic dependency when trying to access this static function in the CentralClientUtils.

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java (hunk 1 above)
Comment (Ibaqu) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java]: Made use of `JvmTarget.JAVA_11.code()`

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java (hunk 2 above)
Comment (Ibaqu) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectCentralUtils.java]: Cyclic dependency happens when we try to use the `CentralUtils` functions.

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/RemotePackageRepositoryException.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package io.ballerina.projects;
+
+/**
+ * A {@code RemotePackageRepositoryException} is thrown for any unrecoverable errors that
```
Comment (Ibaqu) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/RemotePackageRepositoryException.java:4]: @hevayo Created a new exception `RemotePackageRepositoryException`. Is this ok?

**Comment by Ibaqu** [comment]
Closing since this will be handled in a different implementation

**Pull request closed without merging**